plugins/bulk-import/src/components/AddRepositories/AddRepositoriesForm.tsx

```diff
@@ -1,133 +1,112 @@
+    if (Object.keys(dryRunErrors?.errors || {}).length > 0) {
+      formikHelpers.setStatus(dryRunErrors);
+      formikHelpers.setSubmitting(false);
+    } else {
+      formikHelpers.setStatus(dryRunErrors); // to show info messages
+      const submitResult = await mutationCreate.mutateAsync({
+        importJobs: importRepositories,
+      });
+      formikHelpers.setSubmitting(true);
+      const createJobErrors = getJobErrors(submitResult as ImportJobResponse[]);
+      if (Object.keys(createJobErrors?.errors || {}).length > 0) {
+        formikHelpers.setStatus(createJobErrors);
+      } else {
+        navigate(`..`);
+      }
+    }
+
+    formikHelpers.setSubmitting(false);
   };
 
   return (
-    <>
-      <FormControl fullWidth>
-        <div className={styles.body}>
-          {error && (
-            <div style={{ paddingBottom: '10px' }}>
-              <Alert severity="error">
-                <AlertTitle>{error?.title}</AlertTitle>
-                {error?.message}
-              </Alert>
-            </div>
-          )}
-          {/* 
-          // Enable this when ServiceNow approval tool is supported
-          <span className={styles.approvalTool}>
-            <Typography fontSize="16px" fontWeight="500">
-              Approval tool
-            </Typography>
-            <Tooltip
-              placement="top"
-              title="When adding a new repository, it requires approval. Once the PR is approved or the ServiceNow ticket is closed, the repositories will be added to the Catalog page."
-            >
-              <span className={styles.approvalToolTooltip}>
-                <HelpIcon fontSize="small" />
-              </span>
-            </Tooltip>
-            <RadioGroup
-              id="approval-tool"
-              data-testid="approval-tool"
-              row
-              name="approvalTool"
-              value={values.approvalTool}
-              onChange={(_event, value: string) => {
-                setFieldValue('approvalTool', value);
-              }}
-            >
-              <FormControlLabel value="git" control={<Radio />} label="Git" />
-              <FormControlLabel
-                value="servicenow"
-                control={<Radio />}
-                label="ServiceNow"
-                disabled
-              />
-            </RadioGroup>
-          </span> */}
-          <AddRepositoriesTable title="Selected repositories" />
-        </div>
-        <br />
-      </FormControl>
-      <AddRepositoriesFormFooter />
-      {openDrawer && (
-        <PreviewFileSidebar
-          open={openDrawer}
-          onClose={closeDrawer}
-          data={drawerData}
-          repositoryType={values.repositoryType}
-          handleSave={handleSave}
-        />
-      )}
-    </>
+    <DrawerContextProvider>
+      <Formik
+        initialValues={initialValues}
+        enableReinitialize
+        onSubmit={handleSubmit}
+      >
+        <AddRepositories error={generalSubmitError} />
```

Comment:
Maybe you can pass just `mutationCreate.error` here. Or you can convert after `const mutationCreate` with

```tsx
  const generalSubmitError = mutationCreaten.errro ? ({
    message:
      error?.message ||
        `${get(error, 'error.message')}\n${get(error, 'error.response.url')}` ||
        'Failed to create pull request',
    title: error?.name || get(error, 'error.name') || 'Error occured',
  }) : null;
```

```suggestion
        <AddRepositories error={generalSubmitError} />
```

Wdyt?